File: webapp-angular/lib/src/websocket/Request.dart
// Copyright (c) 2017, arthurdeschamps. All rights reserved. Use of this source code
// is governed by a BSD-styles license that can be found in the LICENSE file.

class Request {
  RequestType requestType;
  List<String> topics;

  Request(this.requestType, this.topics);

  Map toJson() {
    Map map = new Map();
    map["requestType"] = _requestTypeToString(requestType);
    map["topics"] = topics;
    return map;
  }

  String _requestTypeToString(RequestType requestType) {
    return requestType.toString().replaceFirst("RequestType.","");
  }

}

/**
 * Type of request for data. "MULTIPLE" for multiple values (stores), "ONE" for a unique value. "ALL" for anything.
 * "SUBSCRIBE" to subscribe to a particular topic.
 */
enum RequestType {
  MULTIPLE,ONE,ALL,SUBSCRIBE
}

File: src/main/java/communications/websocket/Request.java
package communications.websocket;

/**
 * Class that translates a json request to a well formed request. This class must also exists in the client application.
 * @since 1.0
 * @author Arthur Deschamps
 */
class Request {
    private RequestType requestType;
    private String[] topics;

    public Request(RequestType requestType, String[] topics) {
        this.requestType = requestType;
        this.topics = topics;
    }

    public Request() {
    }

    /**
     * Type of request for data. "MULTIPLE" for multiple values (stores), "ONE" for a unique value. "ALL" for anything.
     */
    enum RequestType {
        MULTIPLE,ONE,ALL
    }

    public RequestType getRequestType() {
        return requestType;
    }

    public void setRequestType(RequestType requestType) {
        this.requestType = requestType;
    }

    public String[] getTopics() {
        return topics;
    }

    public void setTopics(String[] topics) {
        this.topics = topics;
    }
}